CMakeLists.txt

```diff
@@ -513,6 +513,13 @@ endif()
 
 include(CMakePackageConfigHelpers)
 
+if(DEFINED OPENTELEMETRY_BUILD_DLL)
```

Comment:
Why not use `BUILD_SHARED_LIBS`?

Reply:
`BUILD_SHARED_LIBS` will build each `add_library` as a DLL which produces a lot of DLL. I'd like to consolidate all the DLLs into 1 with only OpenTelemetry API and selected functions exported.

Reply:
Thanks for the explaination.